LayoutKit/Layouts/SizeLayout.swift

```diff
@@ -88,28 +151,90 @@ public class SizeLayout<V: View>: BaseLayout<V>, Layout {
+                            viewReuseId: String? = nil,
+                            sublayout: Layout? = nil,
+                            config: (V -> Void)? = nil) {
+
+        self.init(minWidth: minSize.width,
+                  minHeight: minSize.height,
+                  alignment: alignment,
+                  flexibility: flexibility,
+                  viewReuseId: viewReuseId,
+                  sublayout: sublayout,
+                  config: config)
+    }
+
+    private static func defaultAlignment(maxWidth maxWidth: CGFloat?, maxHeight: CGFloat?) -> Alignment {
+        return Alignment(vertical: maxHeight == nil ? .fill : .center,
+                         horizontal: maxWidth == nil ? .fill : .center)
+    }
+
+    private static func defaultFlexibility(minWidth minWidth: CGFloat?,
+                                                    maxWidth: CGFloat?,
+                                                    minHeight: CGFloat?,
+                                                    maxHeight: CGFloat?) -> Flexibility {
+        let horizontal = dimensionFlex(min: minWidth, max: maxWidth)
+        let vertical = dimensionFlex(min: minHeight, max: maxHeight)
+        return Flexibility(horizontal: horizontal, vertical: vertical)
+    }
+
+    /// If we have an exact constraint on a dimension then it is inflexible.
+    /// Otherwise, it has default flex.
+    private static func dimensionFlex(min min: CGFloat?, max: CGFloat?) -> Flexibility.Flex {
+        return equals(min, max) ? Flexibility.inflexibleFlex : Flexibility.defaultFlex
+    }
+
+    /// Returns true if left and right are equal within a small tolerance.
+    /// A nil value is not equal to anything else.
+    private static func equals(left: CGFloat?, _ right: CGFloat?) -> Bool {
+        guard let left = left, right = right else {
+            // treat nil != nil
+            return false
         }
+        return abs(left - right) < 0.0001
+    }
+
+    public func measurement(within maxSize: CGSize) -> LayoutMeasurement {
+        // Take the smaller of our configured max size and the given max size for measurement.
+        let availableSize = maxSize.decreasedToSize(CGSize(width: maxWidth ?? .max, height: maxHeight ?? .max))
+
+        // Measure the sublayout if it exists.
+        let sublayoutMeasurement = sublayout?.measurement(within: availableSize)
+        let sublayoutSize = sublayoutMeasurement?.size ?? CGSizeZero
 
-        return LayoutMeasurement(layout: self, size: constrainedSize, maxSize: maxSize, sublayouts: [])
+        // Make sure that our size is in the desired range.
+        let size = sublayoutSize.increasedToSize(CGSize(width: minWidth ?? 0, height: minHeight ?? 0)).decreasedToSize(availableSize)
+
+        let sublayouts = sublayoutMeasurement.map { (measurement) in
+            return [measurement]
+        }
+        return LayoutMeasurement(layout: self, size: size, maxSize: maxSize, sublayouts: sublayouts ?? [])
```

Comment:
consider doing `let sublayouts = [sublayoutMeasurement].flatMap { $0 }`, then you won't need `?? []`